spec/requests/prayers_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+
+RSpec.describe 'Prayer endpoints', :request do
+  include Rack::Test::Methods
+  include ApiSpecHelpers
+
+  let(:app) { Sinatra::Application }
+  let(:user) { create(:user) }
+  let(:headers) do
+    {
+      'P4L-email' => user.email
+    }
+  end
+
+  describe 'POST /prayers' do
+    let!(:resident) { create(:resident, user:) }
+
+    let(:valid_params) do
+      {
+        resident_id: resident.id
+      }
+    end
+
+    it 'requires the email header' do
+      post '/prayers', valid_params, {}
+
+      expect(last_response.status).to eq(400)
+      expect(parsed_response['errors']).to include('Missing param: P4L-email')
+    end
+
+    it 'creates a Prayer' do
+      expect do
+        post '/prayers', valid_params, headers
+      end.to change { Prayer.count }.by(1)
+    end
+
+    it 'returns the next Resident for the User' do
+      bob = create(:resident, user:, position: resident.position + 1)
+
+      post '/prayers', valid_params, headers
+      expect(last_response.status).to eq(201)
+
+      expect(parsed_response).to include(
+        data_object_for(Prayer.last).merge('next_resident_id' => bob.id)
+      )
+    end
+
+    it 'returns an error response for a malformed request' do
```

Comment:
I might change this test a bit because it doesn't test what it says it does. If the request had invalid JSON then I would consider it malformed. This is just an incorrect parameter, but valid JSON.